code/game/objects/items/weapons/landmines.dm

```diff
@@ -196,6 +196,69 @@
 	fragem(src,num_fragments,num_fragments,explosion_size,explosion_size+1,fragment_damage,damage_step,TRUE)
 	qdel(src)
 
+/**
+ * # Door Rigging Landmine
+ *
+ * A landmine that will explode when the door it is attached to opens
+ */
+/obj/item/landmine/frag/door_rigging
+	name = "Door Rigging Landmine"
+	fragment_damage = 20
+
+	///The airlock that we are observing for when it opens, to explode
+	var/obj/machinery/door/airlock/door_rigged
+
+//Prevent this mine to be used like a normal one
+/obj/item/landmine/frag/door_rigging/attack_self(mob/user)
+	to_chat(user, SPAN_ALERT("This landmine is not usable in this way, you need to apply it to a door."))
+	return
+
+/obj/item/landmine/frag/door_rigging/resolve_attackby(atom/A, mob/user, click_parameters)
+	. = ..()
+
+	if(istype(A, /obj/machinery/door/airlock))
+
+		door_rigged = A
+		var/turf/turf_under_door = get_turf(door_rigged)
+
+		//Prevent people from exploding themselves by targeting a door that will open once clicked
+		if(!door_rigged.welded || !door_rigged.density || !istype(turf_under_door) || locate(/obj/item/landmine) in turf_under_door)
+			to_chat(user, SPAN_WARNING("The door is not welded, is open, is already rigged or does not have a turf below it."))
+			door_rigged = null //Clean up the var
+			return
+
+		//Take a little to do this
+		if(!do_after(user, 10 SECONDS, door_rigged))
+			door_rigged = null
+			return
+
+		RegisterSignal(door_rigged, COMSIG_QDELETING, PROC_REF(handle_door_qdel))
+
+		deploy(user)
+		src.forceMove(turf_under_door)
```

Comment:
i don't remember exactly what the proc is, but i think `user.drop_from_inventory(src, turf_under_door)` is better for moving something out of the user's hands, iirc it cleans up some stuff

Reply:
This is already done by deploy()